File: lib_openweather/src/main/java/guepardoapps/library/openweather/controller/IReceiverController.java
package guepardoapps.library.openweather.controller;

import android.content.BroadcastReceiver;
import android.support.annotation.NonNull;

public interface IReceiverController {
    void RegisterReceiver(@NonNull BroadcastReceiver receiver, @NonNull String[] actions);

    void UnregisterReceiver(@NonNull BroadcastReceiver receiver);

    void Dispose();
}

File: app/src/main/java/guepardoapps/lib/openweather/controller/IReceiverController.kt
package guepardoapps.lib.openweather.controller

import android.content.BroadcastReceiver
import android.support.annotation.NonNull

interface IReceiverController {
    fun registerReceiver(@NonNull broadcastReceiver: BroadcastReceiver, @NonNull actions: Array<String>)

    fun unregisterReceiver(@NonNull broadcastReceiver: BroadcastReceiver)

    fun dispose()
}